Get photos script

armanrahman22 opened the pull request.

armanrahman22 committed:
fetch image scripts

armanrahman22 committed:
fetch image scripts

armanrahman22 committed:
fix docker filepath

armanrahman22 committed:
fix docker filepath

c-w committed:
Avoid unauthorized user warnings in MySQL logs

c-w committed:
Replace Azure Queue polling with Celery

c-w committed:
Factor out SQLAlchemy connection string

c-w committed:
Make tests verbose

c-w committed:
Make celery concurrency configurable

c-w committed:
Reduce test timeout

c-w committed:
Add link to algorithms README

c-w committed:
Merge pull request #29 from c-w/update-readme

c-w committed:
Mark non-implemented tests as skipped

c-w committed:
Make processing more reliable

c-w committed:
Add face vector computation processing step

c-w committed:
Update to latest Flask

c-w commented on .gitignore:
```diff
@@ -100,3 +102,5 @@ ENV/
 
 # mypy
 .mypy_cache/
+
```
Nit: trailing spaces

armanrahman22 replied:
fixed!

c-w commented on scripts/common/models/.gitignore:
```diff
@@ -0,0 +1 @@
+# *.pb
```
Nit: commented out line.

armanrahman22 replied:
fixed!

c-w commented on scripts/get_famous_people_list/.dockerignore:
```diff
@@ -0,0 +1,4 @@
+__pycache__/
+.vscode/
+.ipynb_checkpoints/
+.DS_Store
```
Nit: missing newline at end of file

armanrahman22 replied:
fixed!

c-w commented on scripts/common/text_files/country_demonyms.txt:
```diff
@@ -0,0 +1,32 @@
+Syrians
+Turks
+Emiratis
+Yemenis
```
Consider compressing the data files with gzip so that git doesn't try to line-diff them. Using gzip from Python is just about as easy as using normal files (`import gzip; with gzip.open(...) as fobj: ...`)

armanrahman22 replied:
done!

c-w commented on scripts/get_famous_people_list/Dockerfile:
```diff
@@ -0,0 +1,13 @@
+COPY ./get_famous_people_list/requirements.txt /app/requirements.txt
+RUN pip install --upgrade pip
+RUN pip install --no-cache-dir -r /app/requirements.txt
+RUN python -m spacy download en_core_web_lg
```
Consider merging the three RUN layers so that you minimize the number of docker layers and therewith image size. See tip 3 here: https://hackernoon.com/tips-to-reduce-docker-image-sizes-876095da3b34

armanrahman22 replied:
fixed!

c-w commented on scripts/get_famous_people_list/Dockerfile:
```diff
@@ -0,0 +1,13 @@
+WORKDIR /app/get_famous_people_list/src
+
+
+CMD ["python", "get_famous_people_list.py"]
```
Nit: missing newline at end of file

armanrahman22 replied:
fixed!

c-w commented on scripts/get_famous_people_list/requirements.txt:
```diff
@@ -0,0 +1,5 @@
+beautifulsoup4
+spacy
+
+
```
Nit: trailing newlines

armanrahman22 replied:
fixed!

c-w commented on scripts/get_famous_people_list/src/get_famous_people_list.py:
```diff
@@ -0,0 +1,63 @@
+    soup = BeautifulSoup(result.content, 'html.parser')
+
+    demonyms = []
+    with open(demonym_file) as fp:
```
Consider specifying the encoding of the file (`open(demonym_file, encoding='utf-8')`) to ensure the script can run on platforms where utf-8 is not the default encoding (e.g. Windows).

armanrahman22 replied:
fixed!

c-w commented on scripts/get_famous_people_list/src/get_famous_people_list.py:
```diff
@@ -0,0 +1,63 @@
+
+    demonyms = []
+    with open(demonym_file) as fp:
+        demonyms = fp.read().splitlines() 
```
Given that you're using this as a contains check later, consider turning it into a set: `demonyms = {entity.strip() for entity in fp}`

armanrahman22 replied:
fixed!

c-w commented on scripts/get_famous_people_list/src/get_famous_people_list.py:
```diff
@@ -0,0 +1,63 @@
+    """Given a list of pages, gets all people's names
+    
+    Arguments:
+        pages {str[]} -- list of urls to famous people from country pages
```
Nit: instead of comments, consider using Python 3 type annotations which tend to get maintained more than comments. E.g.:

```
from typing import List

def get_people(pages: List[str], output_file: str):
```

armanrahman22 replied:
fixed!

c-w commented on scripts/get_famous_people_list/src/get_famous_people_list.py:
```diff
@@ -0,0 +1,63 @@
+
+    people = []
+    for page in pages:
+        result = requests.get(page)
```
Consider using [grequests](https://pypi.org/project/grequests/) to run the requests in parallel or [aiohttp](https://github.com/aio-libs/aiohttp#client) to run the requests with non-blocking IO.

armanrahman22 replied:
fixed!

c-w commented on scripts/get_famous_people_list/src/get_famous_people_list.py:
```diff
@@ -0,0 +1,63 @@
+
+if __name__== "__main__":
+    pages = get_pages('../../common/text_files/country_demonyms.txt')
+    get_people(pages, '../../common/text_files/famous_people.txt')
```
Nit: missing newline. I will stop commenting on missing newlines and trailing newlines now.

armanrahman22 replied:
fixed!

c-w commented on scripts/get_famous_people_photos/src/get_famous_people_photos.py:
```diff
@@ -0,0 +1,277 @@
+    if len(deduped_urls) > 5:
+        directory = images_directory + person
+        if not os.path.exists(directory):
+            os.makedirs(directory)
```
This will have a race condition in case the directory gets created after the exists check. It's better to create the directory and ignore the exception if it already exists. Python 3.2+ even makes this very simple with the [exist_ok](https://docs.python.org/3/library/os.html#os.makedirs) flag: `makedirs(directory, exist_ok=True)`

armanrahman22 replied:
fixed!

c-w reviewed (approved):
Fine overall. Some of the multiprocessing might have been avoidable with a non-blocking http client so perhaps that's something to investigate for future scripts.

c-w committed:
Merge pull request #30 from c-w/celery

c-w committed:
Move CI runner to makefile

c-w committed:
Ensure that docker can be reached

c-w committed:
Ensure test vectorizer image is available

c-w committed:
Ensure that every make run cleans up after itself

c-w committed:
Make test success required

c-w committed:
Reduce test timeout

Travis pre-fetches the vectorizer docker images so the test should take
on the order of seconds, not minutes.

c-w committed:
Remove the need for explicit clean make recipe

c-w committed:
Add production server recipe

c-w committed:
Use per-test database in mysql

c-w committed:
Separate tests and application code

c-w committed:
Enable code coverage reporting

c-w committed:
Move celery integration to own file

c-w committed:
Return assigned image id in upload image call

c-w committed:
Make API response content types consistent

c-w committed:
Factor out API error messages

c-w committed:
Make RabbitMQ logging consistent

c-w committed:
Merge pull request #31 from c-w/make

armanrahman22 committed:
fixing comments from pull request

armanrahman22 committed:
merging

armanrahman22 committed:
fixes

armanrahman22 committed:
formated file
